interview-questions.md

```diff
@@ -10,48 +10,116 @@ Then, research the question to expand on your answer. Even if you feel you have
+   
   Researched answer:
 
+  . map() creates a new array from calling a function for every array element. map() calls a function once for each element in an array. map() does not execute the function for empty elements.
+  ----- EX -----
+  .map(value, index, array)
+
 
 
 3. What is the difference between map and filter?
 
   Your answer:
 
+  The .map() is used to loop arrays and return new arrays the same length as the original array.
+  
+  The .filter() is also used to loop arrays but it only return the value stated in the if statement. 
+
   Researched answer:
 
+.map() and .filter are higher order functions. Higher order functions take other functions as arguments and/or return another function as an output.
+
+.map() cycle through an array and returns a new array with the updated values.
+EX
+var myArray = [5, 3, 2, 9, 8]
+
+const multiplier = myArray.map(value => value * 3)
+
+console.log(multiplier)
+
+.filter cycle through an array and returns a new array with values that satisfy your stated condition.
+EX
+const luckyNumbers = [85, 7, 5, 0, 45, 77, 9, 6]
+
+const onlyEven = (luckyNumbers) => {
+  
+This example shows the array luckyNumbers being filtered .filter through for numbers that do not have a remainder after being divided by 2.
+The numbers that return are the new array onlyEven and will be console logged for output.
+
+  return luckyNumbers.filter(value => value % 2 === 0)
+
+console.log(return luckyNumbers.filter(value => value % 2 === 0))
+
+OUTPUT: [0, 6]
+
 
 
 4. What is the difference between a function and a method?
 
   Your answer:
 
+  The difference between a function and a method are, functions are the statement being executed. Not sure about method. 
+
   Researched answer:
 
+  Functions are a set statement that perform task or calculates a value. A method contains functions functions stored as object properties.
+  ----- EX ------
+  object = {
+    methodName: function()
+ 
 
 
 5. What is object destructuring?
 
   Your answer:
 
+  Object destructuring is the process of reassigning variables within a scope.
+
   Researched answer:
 
+  The destructuring assignment is a way of assigning variables in Javascript. Destructuring allows you to take something like an array or an object and unpack their values into individual variables.
+  ----- EX ----
+   A sneakers object
+const sneakers = {
+  brand: "Nike",
+  color: "Grey"
+
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
   Your answer:
 
+  I do not know the definition of hoisting, but im sure this is something related to Javascript 
+
   Researched answer:
 
+  Hoisting is JavaScript's default behavior of moving all declarations to the top of the current scope or the current function.
+  ----- EX ------
+   const numbersArray1 = [6, 7, 8, 9, 10]---------- This is hoisting a declaration
+
+const timesThree = numbersArray1.map(value => {
+  return value * 3
+})
+
+console.log(timesThree)
```

Comment:
This is example is close.  An example that is similar to this would be:
`const timesThree = () => {
  return numbersArray1.map(value => {
  return value * 3
  })
}
const numbersArray1 = [6, 7, 8, 9, 10]
console.log(timesThree())`
The variable is declared after the function that is using it.  Javascript has automatically moved that variable above the function so that it can be utilized inside the function.